test/integration/evictions/evictions_test.go

```diff
@@ -161,76 +163,86 @@ func TestConcurrentEvictionRequests(t *testing.T) {
 		}
 	}
 
-	waitToObservePods(t, podInformer, 10)
+	waitToObservePods(t, podInformer, numOfEvictions)
 
 	pdb := newPDB()
 	if _, err := clientSet.Policy().PodDisruptionBudgets(ns.Name).Create(pdb); err != nil {
 		t.Errorf("Failed to create PodDisruptionBudget: %v", err)
 	}
 
-	waitPDBStable(t, clientSet, 10, ns.Name, pdb.Name)
+	waitPDBStable(t, clientSet, numOfEvictions, ns.Name, pdb.Name)
 
-	doneCh := make(chan bool, 10)
-	errCh := make(chan error, 1)
-	// spawn 10 goroutine to concurrently evict the pods
-	for i := 0; i < 10; i++ {
+	doneCh := make(chan bool, numOfEvictions)
+	errCh := make(chan error, 2*numOfEvictions)
+	var wg sync.WaitGroup
+	// spawn numOfEvictions goroutines to concurrently evict the pods
+	for i := 0; i < numOfEvictions; i++ {
+		wg.Add(1)
 		go func(id int, doneCh chan bool, errCh chan error) {
-			evictionName := fmt.Sprintf(podNameFormat, id)
-			eviction := newEviction(ns.Name, evictionName, deleteOption)
-
+			defer wg.Done()
+			podName := fmt.Sprintf(podNameFormat, id)
+			eviction := newEviction(ns.Name, podName, deleteOption)
 			var e error
-			for {
+			if err := wait.PollImmediate(5*time.Second, 60*time.Second, func() (bool, error) {
 				e = clientSet.Policy().Evictions(ns.Name).Evict(eviction)
-				if errors.IsTooManyRequests(e) {
-					time.Sleep(5 * time.Second)
-				} else {
-					break
+				switch {
+				case errors.IsTooManyRequests(e):
+					return false, nil
+				case errors.IsConflict(e):
+					return false, fmt.Errorf("Unexpected Conflict (409) error caused by failing to handle concurrent PDB updates: %v", e)
+				case e == nil:
+					return true, nil
+				default:
+					return false, e
 				}
-			}
-			if e != nil {
-				if errors.IsConflict(e) {
-					e = fmt.Errorf("Unexpected Conflict (409) error caused by failing to handle concurrent PDB updates: %v", e)
+			}); err != nil {
+				errCh <- err
+			} else {
+				_, err = clientSet.Core().Pods(ns.Name).Get(podName, metav1.GetOptions{})
+				switch {
+				case errors.IsNotFound(err):
+					doneCh <- true
```

Comment:
if we return from the goroutine here can we pull out the `e = clientSet.Core().Pods(ns.Name).Delete(podName, deleteOption)` to after the `switch` statement so as to not duplicate the line since we handle both the `default` and `err == nil` cases the same way by deleting the pod and collecting any error returned